nchelpers/__init__.py

```diff
@@ -0,0 +1,405 @@
+        - We want a common interface, i.e., common names, for a selected set of those differently named attributes.
+        - We must avoid shadowing existing properties and methods on a CFDataset (or really, a netCDF4.Dataset) object
+          with the unified names we'd like to use for these metadata properties.
+        - We'd like to present them as properties instead of as a dict, which has uglier syntax
+        How?
+        - Create a property called metadata on CFDataset that is an instance of this class.
+        """
+        # TODO: Make this a singleton
+
+        def __init__(self, dataset):
+            self.dataset = dataset
+
+        _aliases = {
+            'institution': {
+                'CMIP3': 'institute',
+                'CMIP5': 'institute_id',
+            },
+            'model': {
+                'CMIP3': 'source',
+                'CMIP5': 'model_id',
+            },
+            'emissions': {
+                'CMIP3': 'experiment_id',
+                'CMIP5': 'experiment_id',
+            },
+            'run': {
+                'CMIP3': 'realization',
+                'CMIP5': 'parent_experiment_rip',
+            },
+            'project': {
+                'CMIP3': 'project_id',
+                'CMIP5': 'project_id',
+            },
+        }
+
+        def __getattr__(self, alias):
+            project_id = self.dataset.project_id
+            if project_id not in ['CMIP3', 'CMIP5']:
+                raise ValueError("Expected file to have project id of 'CMIP3' or 'CMIP5', found '{}'"
+                                 .format(project_id))
+            if alias not in self._aliases.keys():
+                raise AttributeError("No such unified attribute: '{}'".format(alias))
+            attr = self._aliases[alias][project_id]
+            try:
+                return getattr(self.dataset, attr)
+            except:
+                raise AttributeError("Expected file to contain attribute '{}' but no such attribute exists"
+                                     .format(attr))
+
+    @property
+    def metadata(self):
+        """Prefix for all aliased (common-name) global metadata attributes"""
+        return self.UnifiedMetadata(self)
+
+    @property
+    def is_unprocessed_model_output(self):
+        """True iff the content of the file is unprocessed model output.
+        This allows us to discern between raw amd downscaled model output, for example"""
+        try:
+            self.metadata.model
+        except AttributeError:
+            return False
+        else:
+            return True
+
+    @property
+    def climo_periods(self):
+        """List of those standard climatological periods (see util.standard_climo_periods) that are a subset of the
+        date range in the file."""
+        time_var = self.time_var
+        s_time, e_time = self.time_range
+        return dict([(k, v) for k, v in standard_climo_periods(time_var.calendar).items()
```

Comment:
Also, in your broadcast assignment, you can say ```for k, (start, end) in standard_climo_periods(...``` and then use those aliases to clarify the intent in the conditional below.